src/audio_worklet_node.rs

```diff
@@ -188,44 +247,46 @@ fn process_audio_worklet(env: &Env, processors: &JsObject, args: ProcessorArgume
             let k_worklet_params_cache =
                 env.symbol_for("node-web-audio-api:worklet-params-cache")?;
 
-            let js_inputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_inputs)?;
-
-            for (input_number, input) in inputs.iter().enumerate() {
-                let mut channels = js_inputs.get_element::<JsObject>(input_number as u32)?;
-
-                for (channel_number, channel) in input.iter().enumerate() {
-                    let samples =
-                        float_buffer_to_js(env, channel.as_ptr() as *mut _, channel.len());
-                    channels.set_element(channel_number as u32, samples)?;
-                }
+            // @todo - get from global
+            let render_quantum_size = 128;
+            let mut js_inputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_inputs)?;
+            let mut js_outputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_outputs)?;
+            let mut js_params = processor.get_property::<JsSymbol, JsObject>(k_worklet_params)?;
+            let js_params_cache =
+                processor.get_property::<JsSymbol, JsObject>(k_worklet_params_cache)?;
 
-                // delete remaining channels, if any
-                for i in input.len() as u32..channels.get_array_length().unwrap() {
-                    channels.delete_element(i)?;
-                }
+            // Check JS input and output, and rebuild JS object if layout changed
+            if !check_same_io_layout(&js_inputs, inputs) {
+                let new_js_inputs = rebuild_io_layout(env, js_inputs, inputs, render_quantum_size);
+                // Store new layout in processor
+                processor.set_property(k_worklet_inputs, new_js_inputs)?;
+                // Override js_inputs with new reference
+                js_inputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_inputs)?;
             }
 
-            let js_outputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_outputs)?;
-
-            for (output_number, output) in outputs.iter().enumerate() {
-                let mut channels = js_outputs.get_element::<JsObject>(output_number as u32)?;
+            if !check_same_io_layout(&js_outputs, outputs) {
```

Comment:
I was in doubt if it was possible for output channel counts to change at runtime, but it is indeed possible:
https://webaudio.github.io/web-audio-api/#configuring-channels-with-audioworkletnodeoptions (note at item 4)

Reply:
Yup indeed, I think there might be an issue on the rust side on this point, the `audioworkletnode-channel-count.https.html` wpt test is a good example of what is expected. The test is currently crashing for some reason I didn't really understand, but it seems the rs output length keeps to be 1 even when input is 3